pack/pack.go

```diff
@@ -82,7 +83,12 @@ func encryptFile(encrypter crypto.Encrypter, key []byte, m *xmlenc.Manifest, fil
 	data.KeyInfo = &xmlenc.KeyInfo{}
 	data.KeyInfo.RetrievalMethod.URI = "license.lcpl#/encryption/content_key"
 	data.KeyInfo.RetrievalMethod.Type = "http://readium.org/2014/01/lcp#EncryptedContentKey"
-	data.CipherData.CipherReference.URI = xmlenc.URI(file.Path)
+
+	uri, err := url.Parse(file.Path)
+	if err != nil {
+		return err
+	}
+	data.CipherData.CipherReference.URI = xmlenc.URI(uri.EscapedPath())
```

Comment:
I think that "path normalization" should be applied in `DataForFile(path string)` as well, because there is no guarantee that the `path` string parameter is escaped, and consequently the `CipherData.CipherReference.URI == uri` comparison may fail.

https://github.com/readium/readium-lcp-server/blob/9731d31f4e9c7f9498a86af8f459200c4757711f/xmlenc/encryption.go#L39-L48

Note that `xmlenc.URI` is just a type alias to `string`:
https://github.com/readium/readium-lcp-server/blob/9731d31f4e9c7f9498a86af8f459200c4757711f/xmlenc/encryption.go#L77

Usages of `DataForFile(path string)`:

https://github.com/readium/readium-lcp-server/blob/9731d31f4e9c7f9498a86af8f459200c4757711f/pack/pack.go#L37
=> here, `res.Path` is like `file.Path` (i.e. `epub.Resource.Path`, so must be normalized using `url.Parse(p).EscapedPath()` as well)

https://github.com/readium/readium-lcp-server/blob/9731d31f4e9c7f9498a86af8f459200c4757711f/epub/reader.go#L129
=> here, `file.Name` is `zip.Reader.File.Name`, which may obey different syntactic rules, so once again should be normalized before comparison with `CipherData.CipherReference.URI`

...and test:
https://github.com/readium/readium-lcp-server/blob/9731d31f4e9c7f9498a86af8f459200c4757711f/pack/pack_test.go#L103

___

Note that the proposed change in this PR will affect `CanEncrypt()` too:

https://github.com/readium/readium-lcp-server/blob/9731d31f4e9c7f9498a86af8f459200c4757711f/pack/pack.go#L75-L77
=>
https://github.com/readium/readium-lcp-server/blob/9731d31f4e9c7f9498a86af8f459200c4757711f/epub/epub.go#L89-L93
...and test:
https://github.com/readium/readium-lcp-server/blob/9731d31f4e9c7f9498a86af8f459200c4757711f/pack/pack_test.go#L81

Reply:
Yes Daniel, you're right.
I will fix it